File: lib_base/src/main/java/cn/cbsd/base/event/IBus.java
package cn.cbsd.base.event;

/**
 *
 * @author zhenyanjun
 * @date 2019/1/8 10:56
 */

public interface IBus {

    void register(Object object);

    void unregister(Object object);

    void post(BaseEvent event);

    void postSticky(BaseEvent event);


    abstract class BaseEvent {
        public abstract int getTag();
    }

}

File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/event/IBus.kt
package cn.cbsd.mvplibrary.event

/**
 * Created by wanglei on 2016/12/22.
 */

interface IBus {

    fun register(obj: Any)

    fun unregister(obj: Any)

    fun post(event: AbsEvent)

    fun postSticky(event: AbsEvent)


}
